domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1894,12 +1894,28 @@ public TotaledResults(RowSet<Row> set, Integer estimatedTotal) {
                             Handler<AsyncResult<PostgresClientStreamResult<T>>> replyHandler) {
     // Start a transaction that we need to close.
     // If a transaction is already running we don't need to close it.
-    final Transaction transaction = connection.tx == null ? connection.conn.begin() : null;
+    if (connection.tx == null) {
+      connection.conn.begin().onComplete((AsyncResult<Transaction> trans) -> {
```

Comment:
`onComplete` and `if` can be replaced by `onSuccess(trans -> ...).onFailure(e -> ...)` for readability.